control-plane/api-gateway/binding/route_binding.go

```diff
@@ -20,6 +20,11 @@ func (r *Binder) bindRoute(route client.Object, boundCount map[gwv1beta1.Section
 	// on non-enterprise installations
 	routeConsulKey := r.config.Translator.NonNormalizedConfigEntryReference(entryKind(route), client.ObjectKeyFromObject(route))
 	filteredParents := filterParentRefs(r.key, route.GetNamespace(), getRouteParents(route))
+	filteredParentStatuses := filterParentRefs(r.key, route.GetNamespace(),
+		common.ConvertSliceFunc(getRouteParentsStatus(route), func(parentStatus gwv1beta1.RouteParentStatus) gwv1beta1.ParentReference {
+			return parentStatus.ParentRef
+		}),
+	)
```

Comment:
this was added due to the fact that we now include in the routes we pass in to be reconciled with the gateway the routes that just reference the gateway in their statuses (in order to purge them if necessary)